Source/ACE.Server/WorldObjects/ManaStone.cs

```diff
@@ -58,8 +58,23 @@ public void SetUiEffect(Player player, UiEffects effect)
 
         public void HandleActionUseOnTarget(Player player, WorldObject target)
         {
+            WorldObject invTarget;
```

Comment:
Instead of having to do another lookup to see if the item is in inventory, it would be more efficient to add another arg to this function "bool targetIsInInventory".

Reply:
The outer FindObject() isn't being picky about whether the target is inventory or not either.  The ManaStone branch is the only one so far that really cares about the item being inventory, so I was placing the specific check where the specific matters.